src/main/java/bio/terra/service/snapshotbuilder/SnapshotBuilderService.java

```diff
@@ -83,7 +83,9 @@ public SnapshotBuilderGetConceptsResponse getConceptChildren(
       UUID datasetId, Integer conceptId, AuthenticatedUserRequest userRequest) {
     Dataset dataset = datasetService.retrieve(datasetId);
     TableNameGenerator tableNameGenerator = getTableNameGenerator(dataset, userRequest);
-    String cloudSpecificSql = buildConceptChildrenQuery(conceptId, tableNameGenerator);
+    String cloudSpecificSql =
+        buildConceptChildrenQuery(
+            conceptId, tableNameGenerator, CloudPlatformWrapper.of(dataset.getCloudPlatform()));
```

Comment:
Maybe we should wait and see how many cases we'll have like this, but we could extend `TableNameGenerator` to also return the cloud platform since it needs to know that too.

Reply:
Semantically I would maybe rename it if we were doing that, but I like the idea of this change, especially since tableNameGeneration is mostly a function of cloud platform anyway